azure/deploymentframeworks/arm/output.ftl

```diff
@@ -37,19 +37,63 @@
     }]
 [/#function]
 
-[#macro armOutput name type value condition=""]
-    [@mergeWithJsonOutput
-        name="outputs"
-        content=
-            {
-                name : {
-                    "type" : type,
-                    "value" : value
-                } +
-                attributeIfContent("condition", condition)
-            }
-    /]
-[/#macro]
+[#function getArmOutput name type value condition=""]  
+    [#return {
+        name : {
+            "type" : type,
+            "value" : value
+        } +
+        attributeIfContent("condition", condition)
+    }]
+[/#function]
+
+[#function constructArmOutputsFromMappings id name scope mappings=[]]
+    [#local result = {}]
+    [#list mappings as attributeType,attributes]
+        [#local dataType = getOutputMappingDataType(attributeType)]
+        [#list attributes as attributeName,attributeValue]
+
+            [#switch scope]
+
+                [#case "subscription"]
+                [#case "resourceGroup"]
+                    [#if attributeValue == "id"]
+                        [#local typeFull = getAzureResourceProfile(AZURE_DEPLOYMENT_RESOURCE_TYPE).type]
+                        [#local propertySections = attributeValue?split(".")]
+                        [#local outputName = id]
+                        [#local value = getReference(id, name, typeFull, "", "", "", true, "outputs." + outputName + ".value")]
+                    [#else]
+                        [#local typeFull = getAzureResourceProfile(AZURE_DEPLOYMENT_RESOURCE_TYPE).type]
```

Comment:
Will this handle a missing resource profile?

Reply:
Updated in 53c149e to fail elegantly